src/components/action-popover/action-popover.test.tsx

```diff
@@ -0,0 +1,2303 @@
+  expect(screen.getByRole("button", { name: "first item" })).toBeFocused();
+});
+
+test("an error is thrown, with appropriate error message, if invalid children are used", () => {
+  const globalConsoleSpy = jest
+    .spyOn(global.console, "error")
+    .mockImplementation(() => {});
+
+  expect(() => {
+    render(
+      <ActionPopover>
+        <ActionPopoverItem onClick={() => {}}>Item</ActionPopoverItem>
+        Invalid children
+        <p>invalid children</p>
+      </ActionPopover>
+    );
+  }).toThrow(
+    "ActionPopover only accepts children of type `ActionPopoverItem`" +
+      " and `ActionPopoverDivider`."
+  );
+
+  globalConsoleSpy.mockRestore();
+});
+
+describe("when an item has a submenu with default (left) alignment", () => {
+  it("renders the appropriate icon with the correct alignment", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    render(
+      <ActionPopover>
+        <ActionPopoverItem>example item 1</ActionPopoverItem>
+        <ActionPopoverItem>example item 2</ActionPopoverItem>
+        <ActionPopoverItem
+          submenu={
+            <ActionPopoverMenu>
+              <ActionPopoverItem>submenu item 1</ActionPopoverItem>
+              <ActionPopoverItem>submenu item 2</ActionPopoverItem>
+            </ActionPopoverMenu>
+          }
+        >
+          example item with submenu
+        </ActionPopoverItem>
+      </ActionPopover>
+    );
+
+    await user.click(screen.getByRole("button"));
+
+    // there are 2 elements with data-role="icon", the other being the main menu button.
+    // Unfortunately there seems to be no way with the RTL options for getByTestId to select the correct one, so we have to index
+    // into the array of all such elements
+    const submenuIcon = screen.getAllByTestId("icon")[1];
+    expect(submenuIcon).toHaveStyleRule(
+      "content",
+      `"${iconUnicodes.chevron_left_thick}"`,
+      { modifier: "&::before" }
+    );
+    expect(submenuIcon).toHaveStyle({ left: "-5px" });
+  });
+
+  it("opens the submenu on mouseenter", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    render(
+      <ActionPopover>
+        <ActionPopoverItem>example item 1</ActionPopoverItem>
+        <ActionPopoverItem>example item 2</ActionPopoverItem>
+        <ActionPopoverItem
+          submenu={
+            <ActionPopoverMenu>
+              <ActionPopoverItem>submenu item 1</ActionPopoverItem>
+              <ActionPopoverItem>submenu item 2</ActionPopoverItem>
+            </ActionPopoverMenu>
+          }
+        >
+          example item with submenu
+        </ActionPopoverItem>
+      </ActionPopover>
+    );
+
+    await user.click(screen.getByRole("button"));
+
+    expect(
+      screen.queryByRole("button", { name: "submenu item 1" })
+    ).not.toBeInTheDocument();
+
+    // move mouse over the item with the submenu
+    await user.pointer({
```

Comment:
question: out of interest is there a reason why we're using `pointer` instead of `hover`?

Reply:
yes - that I didn't even think of it 😂 🤦 

I'm actually not 100% sure if this is what `hover` is really for, although it probably is - I'll give it a go.